### PR TITLE
show eror banner for field errors

### PR DESCRIPTION
**Pull request by anipendakur-okta**
## Description:
Pls refer to the video below.
Also this slack thread for more info - https://okta.slack.com/archives/C9AJXB5GC/p1622850598351100

## PR Checklist

- [X] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [X] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:
https://okta.box.com/s/kw5xpnqjeiquode9ljhbdcts7naetlto

### Issue:

- [OKTA-392962](https://oktainc.atlassian.net/browse/OKTA-392962)

### CODE REVIEW

**Comment by anipendakur-okta** [comment]
Note: Once the changes are okay, I will add tests.

**Review by haishengwu-okta — APPROVED** [review]
🚢  after adding tests.

File: src/v2/controllers/FormController.js [hunk 1]
```diff
@@ -248,8 +248,7 @@ export default Controller.extend({
 
     if (IonResponseHelper.isIonErrorResponse(error)) {
       const convertedErrors = IonResponseHelper.convertFormErrors(error);
-      const showBanner = convertedErrors.responseJSON.errorCauses.length ? false : true;
-      model.trigger('error', model, convertedErrors, showBanner);
+      model.trigger('error', model, convertedErrors, true);
```
Comment (haishengwu-okta) [inline src/v2/controllers/FormController.js:251]: nit:
looks like the `if/elseif` block can just create error object and invoke `model.trigger('error', ..` after.

Comment (anipendakur-okta) [reply src/v2/controllers/FormController.js]: Fixed

File: test/testcafe/spec/EnrollAuthenticatorPasswordView_spec.js [hunk 2]
```diff
@@ -70,7 +77,19 @@ test('should succeed when fill same value', async t => {
     .eql('http://localhost:3000/app/UserHome?stateToken=mockedStateToken123');
 });
 
-test('should have the correct reqiurements', async t => {
+test.requestHooks(errorMock)('should show a callout when server-side field errors are received', async t => {
```
Comment (pradeepdewda-okta) [inline test/testcafe/spec/EnrollAuthenticatorPasswordView_spec.js:80]: add another test for client side validation ?

**Review by pradeepdewda-okta — APPROVED** [review]

File: test/testcafe/spec/EnrollAuthenticatorPasswordView_spec.js (hunk 2 above)
Comment (anipendakur-okta) [reply test/testcafe/spec/EnrollAuthenticatorPasswordView_spec.js:80]: I have added above - https://github.com/okta/okta-signin-widget/pull/2028/files#diff-d02cad9b7fcfc4a5bd52981fb90019e6a71df9265f3fb2b7303a35979c92d085R62

**Comment by anipendakur-okta** [comment]
Note - Bacon `PASS` on downstream build.

**Pull request closed without merging**